src/parser/core/modules/Combos.tsx

```diff
@@ -0,0 +1,252 @@
+// If you can make it through this entire file without hitting semantic saturation of the word "combo", hats off to you. IT DOESN'T LOOK REAL ANYMORE.
+
+import {i18nMark, Plural, Trans} from '@lingui/react'
+import {RotationTable} from 'components/ui/RotationTable'
+import {getAction} from 'data/ACTIONS'
+import {AbilityEvent, CastEvent} from 'fflogs'
+import _ from 'lodash'
+import Module, {dependency} from 'parser/core/Module'
+import DISPLAY_ORDER from 'parser/core/modules/DISPLAY_ORDER'
+import Suggestions, {SEVERITY, Suggestion} from 'parser/core/modules/Suggestions'
+import Timeline from 'parser/core/modules/Timeline'
+import React from 'react'
+
+const DEFAULT_GCD = 2.5
+const GCD_TIMEOUT_MILLIS = 12000
+const ISSUE_TYPENAMES = {
+	uncomboed: <Trans id="core.combos.issuetypenames.uncomboed">Uncomboed</Trans>,
+	combobreak: <Trans id="core.combos.issuetypenames.combobreak">Broken Combo</Trans>,
+}
+
+export interface ComboEvent extends AbilityEvent {
+	type: 'combo'
+}
+
+export interface ComboIssue {
+	type: 'uncomboed' | 'combobreak'
+	context: CastEvent[]
+	event: CastEvent
+}
+
+export default class Combos extends Module {
+	static handle = 'combos'
+	static title = 'Combo Issues'
+	static i18n_id = i18nMark('core.combos.title') // tslint:disable-line:variable-name
+	static displayOrder = DISPLAY_ORDER.COMBOS
+
+	// This should be redefined by subclassing modules; the default is the basic 'Attack' icon
+	static suggestionIcon = 'https://xivapi.com/i/000000/000405.png'
+	'constructor': typeof Combos // Allow for `this.constructor` to be accessed
+
+	@dependency protected suggestions!: Suggestions
+	@dependency private timeline!: Timeline
+
+	private lastGcdTime = this.parser.fight.start_time
+	private currentComboChain: CastEvent[] = []
+	private issues: ComboIssue[] = []
+
+	protected init() {
+		this.addHook('cast', {by: 'player'}, this.onCast)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private get lastComboEvent(): CastEvent | null {
+		return _.last(this.currentComboChain) || null
+	}
+
+	private get lastAction() {
+		const lastComboEvent = this.lastComboEvent
+		if (!lastComboEvent) {
+			return null
+		}
+
+		return getAction(lastComboEvent.ability.guid).id
```

Comment:
`getDataBy(ACTIONS, 'id', lastComboEvent.ability.guid)`

but like, guid === id. can just return the guid straight

Reply:
true...